labelbox/client.py

```diff
@@ -1211,3 +1212,27 @@ def _format_failed_rows(rows: List[str],
                     "Timed out waiting for get_data_rows_for_global_keys job to complete."
                 )
             time.sleep(sleep_time)
+
+    def get_slice(self, slice_id) -> CatalogSlice:
```

Comment:
The return type should be `Slice` here, so that we prepare ourselves for ModelSlice later

Reply:
Hmm, in case of we also have `ModelSlice`, isn't the return type should be `Union[CatalogSlice, ModelSlice]` ?

Reply:
Will both kinds of slice objects always be exactly the same and support the same interface and the type is only used to indicate the source?

Reply:
They are probably not 100% the same. I don't think it makes sense to have `get_data_row_ids` in a `ModelSlice`, right? The documents in the Model ES are annotation groups, so when you add filters on the Model UI you filter annotation groups. Maybe I'm wrong here, I don't know how Models work.

Reply:
Yeah there will potentially be differences. So this will prob return the union of the objects (in which case, how you have it now is fine). Although this might be confusing because you call get_slice and have to explicitly check what kind of object you have to know which functions you can use. Or we could create a dedicated function for each which is annoying but very explicit.

Reply:
Gotcha, yeah `Union[CatalogSlice, ModelSlice]` is better. We can leave this as is, and change the return type when we introduce ModelSlice

Reply:
I like the idea better to have 2 functions `get_slice(id) -> CatalogSlice` and `get_model_slice(id) -> ModelSlice`

Reply:
I'm happy with either option. The only thing that I would ask is that if you choose to have two functions they should be ```get_catalog_slice(id) -> CatalogSlice and get_model_slice(id) -> ModelSlice```
(rename `get_slice` to `get_catalog_slice`)

Reply:
Ok, I'll go with `get_catalog_slice(id) -> CatalogSlice`